store/storage_manager_builder.go

```diff
@@ -118,17 +120,35 @@ func (smb *StorageManagerBuilder) Build(ctx context.Context) (*Manager, error) {
 		return nil, fmt.Errorf("dispersal backend is set to V1, but V1 backend is not enabled")
 	}
 
+	var kzgVerifier *kzgverifier.Verifier
+	// there are two cases in which we need to construct the kzgVerifier:
+	// 1. V1
+	// 2. V2, when validator retrieval is enabled
```

Comment:
why dont we need it when retrieving from relays?

Reply:
The verifier is necessary for validator retrieval, to check individual chunks from validator nodes. Retrieval from a relay is simpler, since you get the fully formed blob, and just need to check the cert.

Reply:
for the relay, you also need to check that the blob matches the commitment in the cert right?

Reply:
Yes, but this uses the `Prover`, as opposed to the `Verifier`. This is because we "verify" the commitment by generating our own, and then comparing.